pkg/apis/v1beta1/nodepool.go

```diff
@@ -194,7 +194,7 @@ type NodePool struct {
 // 1. A field changes its default value for an existing field that is already hashed
 // 2. A field is added to the hash calculation with an already-set value
 // 3. A field is removed from the hash calculations
-const NodePoolHashVersion = "v1"
+const NodePoolHashVersion = "v2"
```

Comment:
This is assuming that we have a release with the previous version, right?

Reply:
Yes, defaulting on the nodeClassRef must come after the hash versioning changes have been released

Reply:
Can we add this as a `do-not-merge` in the PR description? I want to make sure I don't approve this and accidentally merge it before we have our ducks in a row.

Reply:
Also, does this really need to be in two different releases? I can imagine that a user upgrades from version from before the hash versioning changes were released to this version with a v2, do we have that case covered?

Reply:
Discussed offline. We will be doing a patch release to each version of the v1beta1 to include drift hash versioning changes.